process-engine/consumer_api_core
# ♻️ Refactor socket io handling

**Changes:**

1. Implement the `removeSubscription` UseCase, which allows a user to unsubscribe from a notification.
2. Implement `subscribeOnce` flag for all notification Subscriptions. This works pretty much like the `subscribeOnce` UseCase for the EventAggregator.
3. Refactor all Notification subscription functions so that they return the created subscriptions.

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/#135
Part of https://github.com/process-engine/process_engine_runtime/issues/#157
Part of https://github.com/process-engine/process_engine_runtime/issues/#206

PR: #83

## How can others test the changes?

- Create some subscriptions for notifications
- Trigger these notifications to receive them
- Unsubscribe from a notification
- Trigger the notification again
- Notice that the callback you provided for that notification will no longer get triggered

## PR-Checklist

Please check the boxes in this list after submitting your PR:
